### PR TITLE
Folder open issue

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Added acceptance test to assert presence of share indicator for shared/received shares on different pages on the webUI

### CODE REVIEW

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 1]
```diff
@@ -659,3 +659,193 @@ Feature: Sharing files and folders with internal users
...
+      | block-aligned.txt                     |
+      | testavatar.png                        |
+      | testapp.zip                           |
+    When the user opens folder "simple-empty-folder" using the webUI
+    And the user uploads file "new-lorem.txt" using the webUI
+    And the user creates a folder with the name "XYZ" using the webUI
+    Then file "new-lorem.txt" should have share indicator on the webUI
+    And folder "XYZ" should have share indicator on the webUI
+    When the user browses to the shared-with-others page
+    Then following resources should not have share indicators on the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
    Then the following resources should not have share indicators on the webUI
```
Adjust the step definition

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 2]
```diff
@@ -659,3 +659,193 @@ Feature: Sharing files and folders with internal users
...
+      | username |
+      | user1    |
+      | user2    |
+    And user "user2" has logged in using the webUI
+    When the user shares folder "simple-folder" with user "User One" using the webUI
+    And the user shares file "lorem.txt" with user "User One" using the webUI
+    Then folder "simple-folder" should not have share indicator on the webUI
+    And file "lorem.txt" should not have share indicator on the webUI
+    When the user opens folder "simple-folder" using the webUI
+    Then following resources should have share indicators on the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
    Then the following resources should have share indicators on the webUI
```

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 3]
```diff
@@ -659,3 +659,193 @@ Feature: Sharing files and folders with internal users
...
+    When the user opens folder "simple-empty-folder" using the webUI
+    And the user uploads file "new-lorem.txt" using the webUI
+    And the user creates a folder with the name "XYZ" using the webUI
+    Then file "new-lorem.txt" should have share indicator on the webUI
+    And folder "XYZ" should have share indicator on the webUI
+    When the user browses to the shared-with-others page
+    Then following resources should not have share indicators on the webUI
+      | simple-folder                   |
+
+  Scenario: user receives a share from another user and view that received share in files page and shared-wih-you page when auto-accept-share is enabled
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
  Scenario: user receives a share from another user and views that received share in files page and shared-wih-you page when auto-accept-share is enabled
```

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 4]
```diff
@@ -659,3 +659,193 @@ Feature: Sharing files and folders with internal users
...
+    Given the setting "Automatically accept new incoming local user shares" in the section "Sharing" has been enabled
+    And these users have been created with default attributes and skeleton files:
+      | username |
+      | user1    |
+      | user2    |
+    And user "user1" has logged in using the webUI
+    And the user has shared folder "simple-folder" with user "User Two" using the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: Use API "given" steps to make the share, then you do not have to login and re-login

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 5]
```diff
@@ -659,3 +659,193 @@ Feature: Sharing files and folders with internal users
...
+    When the user browses to the shared-with-others page
+    And the user opens folder "simple-folder (2)" using the webUI
+    Then following resources should have share indicators on the webUI
+      | simple-empty-folder                   |
+      | lorem.txt                             |
+      | block-aligned.txt                     |
+      | testavatar.png                        |
+      | testapp.zip                           |
+
+  Scenario: shares a sub folder and view the received share form files page and shared-with-you page
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
  Scenario: share a sub folder and view the received share from files page and shared-with-you page
```

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 6]
```diff
@@ -659,3 +659,193 @@ Feature: Sharing files and folders with internal users
...
+    And user "user1" has uploaded file "filesForUpload/lorem.txt" to "simple-folder/simple-empty-folder/lorem.txt"
+    And user "user1" has shared folder "/simple-folder/simple-empty-folder" with user "user2"
+    And user "user2" has logged in using the webUI
+    When the user opens folder "simple-empty-folder (2)" using the webUI
+    Then file "lorem.txt" should not have share indicator on the webUI
+    When the user browses to the shared-with-you page
+    And the user opens folder "simple-empty-folder (2)" using the webUI
+    Then file "lorem.txt" should not have share indicator on the webUI
+
+  Scenario: shares a sub folder and view the shared resource form files page and shared-with-others page
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]:
```suggestion
  Scenario: share a sub folder and view the shared resource from files page and shared-with-others page
```

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature (hunk 1 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: :rocket:

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature (hunk 2 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: :rocket:

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature (hunk 3 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: :rocket:

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature (hunk 4 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: :rocket:

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature (hunk 5 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: :rocket:

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature (hunk 4 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: I've a question...
If I use API "given" steps to create shares, then the "when" part in the scenario would be only `to open` those shared resources. Is that OK?

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 7]
```diff
@@ -1601,4 +1601,68 @@ public function tearDownScenario() {
...
+	 * @param string $filename name of resource
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function folderShouldHaveShareIndicatorOnTheWebUI($fileOrFolder, $filename) {
+		$isMarked = $this->filesPage->isShareIndicatorPresent(
+			$filename
+		);
+		Assert::assertEquals(true, $isMarked, "Expected: " . $fileOrFolder . " to be marked as shared but found: " . \strval($isMarked));
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]: to long line

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 8]
```diff
@@ -1601,4 +1601,68 @@ public function tearDownScenario() {
...
+	 * @param string $filename name of resource
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function folderShouldNotHaveShareIndicatorOnTheWebUI($fileOrFolder, $filename) {
+		$isMarked = $this->filesPage->isShareIndicatorPresent(
+			$filename
+		);
+		Assert::assertEquals(false, $isMarked, "Expected: " . $fileOrFolder . " not to be marked as shared but found: " . \strval($isMarked));
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]: to long line

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 9]
```diff
@@ -1601,4 +1601,68 @@ public function tearDownScenario() {
...
+	/**
+	 * @Then /^(file|folder) "([^"]*)" should not have share indicator on the webUI$/
+	 *
+	 * @param string $fileOrFolder
+	 * @param string $filename name of resource
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function folderShouldNotHaveShareIndicatorOnTheWebUI($fileOrFolder, $filename) {
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]:
```suggestion
	public function ressourceShouldNotHaveShareIndicatorOnTheWebUI($fileOrFolder, $filename) {
```

File: tests/acceptance/features/bootstrap/WebUISharingContext.php [hunk 10]
```diff
@@ -1601,4 +1601,68 @@ public function tearDownScenario() {
...
+		$elements = $this->featureContext->simplifyArray($elementRows);
+		foreach ($elements as $element) {
+			$this->folderShouldHaveShareIndicatorOnTheWebUI(null, $element);
+		}
+	}
+
+	/**
+	 * @Then the following resources should not have share indicators on the webUI
+	 *
+	 * @param TableNode $resourceTable table headings: must be: |name|
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUISharingContext.php]: that is not true, header is not needed

File: tests/acceptance/features/lib/FilesPage.php [hunk 11]
```diff
@@ -399,4 +400,17 @@ public function browseToFolder($folderName) {
...
 	}
+
+	/**
+	 * checks whether given resource is marked as shared or not
+	 *
+	 * @param $fileName
+	 *
+	 * @return bool
+	 */
+	public function isShareIndicatorPresent($fileName) {
```
Comment (individual-it) [inline tests/acceptance/features/lib/FilesPage.php]: move the searching of the indicator to fileRow page object
to find that specific file row use `$this->findFileRowByName()`

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 12]
```diff
@@ -660,6 +660,7 @@ Feature: Sharing files and folders with internal users
     When the user shares file "lorem.txt" with user "User Three" using the webUI
     Then as "user3" file "lorem.txt" should exist
 
+<<<<<<< HEAD
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: ??????

File: tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature [hunk 13]
```diff
@@ -368,8 +368,8 @@ Feature: Share by public link
     And the user opens folder "simple-folder" using the webUI
     And the user creates a new public link for file "lorem.txt" using the webUI
     And the user browses to the shared-by-link page
-    Then file "lorem.txt" with path "" should be listed in the shared with others page on the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature]: don't change this existing test if its still valid

File: tests/acceptance/features/lib/SharedByLinkPage.php [hunk 14]
```diff
@@ -104,6 +114,25 @@ public function __construct(
...
+	/**
+	 * @param string $name
+	 * @param string $path
+	 * @param Session $session
+	 *
+	 * @return NodeElement
+	 *
+	 * @throws ElementNotFoundException
+	 */
+	public function findFileRowByNameAndPath($name, $path, Session $session) {
```
Comment (individual-it) [inline tests/acceptance/features/lib/SharedByLinkPage.php]: I don't think we should need this function

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 15]
```diff
@@ -677,3 +678,192 @@ Feature: Sharing files and folders with internal users
...
+    And the user uploads file "new-lorem.txt" using the webUI
+    And the user creates a folder with the name "XYZ" using the webUI
+    Then file "new-lorem.txt" should have share indicator on the webUI
+    And folder "XYZ" should have share indicator on the webUI
+    When the user browses to the shared-with-others page
+    Then the following resources should not have share indicators on the webUI
+      | simple-folder                   |
+
+  Scenario: user receives a share from another user and views that received share in files page and shared-wih-you page when auto-accept-share is enabled
+    Given the setting "Automatically accept new incoming local user shares" in the section "Sharing" has been enabled
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: I think this is default, so should not need it

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 16]
```diff
@@ -677,3 +678,192 @@ Feature: Sharing files and folders with internal users
...
+    When the user opens folder "simple-folder (2)" using the webUI
+    Then the following resources should not have share indicators on the webUI
+      | simple-empty-folder                   |
+      | lorem.txt                             |
+      | block-aligned.txt                     |
+      | testavatar.png                        |
+      | testapp.zip                           |
+
+  Scenario: user creates a public link for a folder and views the shared folder from files page, shared-with-others page
+    and shared-by-link page
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: why is that in a new line?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/lib/SharedByLinkPage.php (hunk 14 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/lib/SharedByLinkPage.php]: I've reverted everything mess i'd made as this PR is not really concerned about this!
But we've to look into it some other time!

File: tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature (hunk 13 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature]: I've reverted everything back to as it was!
:rocket:

File: tests/acceptance/features/bootstrap/WebUISharingContext.php (hunk 7 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/WebUISharingContext.php]: condensed!
:rocket:

File: tests/acceptance/features/bootstrap/WebUISharingContext.php (hunk 9 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/WebUISharingContext.php]: done
:rocket:

File: tests/acceptance/features/bootstrap/WebUISharingContext.php (hunk 8 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/WebUISharingContext.php]: shortened!
:rocket:

File: tests/acceptance/features/bootstrap/WebUISharingContext.php (hunk 10 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/WebUISharingContext.php]: yup!
:rocket:

**Comment by individual-it** [comment]
closing in favour of  #36534

**Pull request closed without merging**